full-service/src/service/sync.rs

```diff
@@ -51,12 +55,111 @@ pub struct SyncThread {
     stop_requested: Arc<AtomicBool>,
 }
 
+pub struct WebhookThread {
+    /// The main sync thread handle.
+    join_handle: Option<thread::JoinHandle<()>>,
+
+    /// Stop trigger, used to signal the thread to terminate.
+    stop_requested: Arc<AtomicBool>,
+}
+
+impl WebhookThread {
```

Comment:
🆕  This is the new Webhook Thread that checks the shared `AtomicUsize` `num_txos_synced` and fires the webhook callback